earth_enterprise/src/fusion/autoingest/StorageManager.h

```diff
@@ -355,5 +360,22 @@ bool StorageManager<AssetType>::SaveDirtyToDotNew(
   return true;
 }
 
+template<class AssetType>
+bool StorageManager<AssetType>::DetermineIfPrune() {
+  std::lock_guard<std::recursive_mutex> lock(storageMutex);
+  float cacheSize = (float)CacheSize();
+  float dirtySize = (float)DirtySize();
+  //uint32 diff;
+  //float percent;
+
+  //if (cacheSize == 0) { return true; }
+  //diff = cacheSize - dirtySize;
+  //percent = ((float)diff / (float)cacheSize) * 100;
+  //notify(NFY_WARN, "Percent Diff: %f", percent);
+  //return (percent > maxPurgePercent);
+  notify(NFY_WARN, "CacheSize: %f, DirtySize: %f, Percent: %f", cacheSize, dirtySize, (cacheSize * maxPurgePercent));
+  return (dirtySize < (cacheSize * maxPurgePercent));
```

Comment:
Just so this is clear, this changes the meaning of maxPurgePercent compared to your old code. If maxPurgePercent was 0.75, the old code would stop purging when the dirty map got to be 25% of the cache (if I'm reading the code correctly). With the updated code, it will stop purging when the dirty map gets to be 75% of the cache. Either is fine; we just have to make sure we document it correctly when we get to that point.